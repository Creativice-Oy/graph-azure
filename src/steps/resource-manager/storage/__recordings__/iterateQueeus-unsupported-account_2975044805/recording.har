{
  "log": {
    "_recordingName": "iterateQueeus-unsupported-account",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "a083439df9cb90ab0ef889609d0aa995",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 172,
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "name": "accept-charset",
              "value": "utf-8"
            },
            {
              "name": "client-request-id",
              "value": "f2134eb0-6cde-4b38-a5ef-8a0e4799f4b1"
            },
            {
              "name": "return-client-request-id",
              "value": "true"
            },
            {
              "name": "x-client-sku",
              "value": "Node"
            },
            {
              "name": "x-client-ver",
              "value": "0.1.28"
            },
            {
              "name": "x-client-os",
              "value": "darwin"
            },
            {
              "name": "x-client-cpu",
              "value": "x64"
            },
            {
              "name": "host",
              "value": "login.microsoftonline.com"
            },
            {
              "name": "content-length",
              "value": 172
            }
          ],
          "headersSize": 414,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [
            {
              "name": "api-version",
              "value": "1.0"
            }
          ],
          "url": "https://login.microsoftonline.com/992d7bbe-b367-459c-a10f-cf3fd16103ab/oauth2/token?api-version=1.0"
        },
        "response": {
          "bodySize": 1450,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 1450,
            "text": "{\"token_type\":\"Bearer\",\"expires_in\":\"3599\",\"ext_expires_in\":\"3599\",\"expires_on\":\"1603131233\",\"not_before\":\"1603127333\",\"resource\":\"https://management.azure.com/\",\"access_token\":\"[REDACTED]\"}"
          },
          "cookies": [
            {
              "expires": "2020-11-18T17:13:53.000Z",
              "httpOnly": true,
              "name": "fpc",
              "path": "/",
              "sameSite": "None",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "x-ms-gateway-slice",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "stsservicecookie",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "cache-control",
              "value": "no-store, no-cache"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "expires",
              "value": "-1"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "p3p",
              "value": "CP=\"DSP CUR OTPi IND OTRi ONL FIN\""
            },
            {
              "name": "client-request-id",
              "value": "f2134eb0-6cde-4b38-a5ef-8a0e4799f4b1"
            },
            {
              "name": "x-ms-request-id",
              "value": "1635ae5c-2a2c-41f5-96ac-3f5fc74cae00"
            },
            {
              "name": "x-ms-ests-server",
              "value": "2.1.11140.11 - SAN ProdSlices"
            },
            {
              "name": "x-ms-clitelem",
              "value": "1,0,0,,"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "date",
              "value": "Mon, 19 Oct 2020 17:13:52 GMT"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "content-length",
              "value": "1450"
            }
          ],
          "headersSize": 782,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2020-10-19T17:13:52.943Z",
        "time": 521,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 521
        }
      },
      {
        "_id": "5b659db4e0ea98de8227750eb75526ee",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept-language",
              "value": "en-US"
            },
            {
              "_fromType": "array",
              "name": "x-ms-client-request-id",
              "value": "5ca0f092-4f07-4543-a872-f17bf30c0632"
            },
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "ms-rest-js/2.0.7 Node/v14.9.0 OS/(x64-Darwin-19.6.0)"
            },
            {
              "_fromType": "array",
              "name": "cookie",
              "value": ""
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "management.azure.com"
            }
          ],
          "headersSize": 1676,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "api-version",
              "value": "2016-06-01"
            }
          ],
          "url": "https://management.azure.com/subscriptions?api-version=2016-06-01"
        },
        "response": {
          "bodySize": 358,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 358,
            "text": "{\"value\":[{\"id\":\"/subscriptions/d3803fd6-2ba4-4286-80aa-f3d613ad59a7\",\"authorizationSource\":\"RoleBased\",\"subscriptionId\":\"d3803fd6-2ba4-4286-80aa-f3d613ad59a7\",\"displayName\":\"nick-dowmon-gmail-graph-azure-subscription\",\"state\":\"Enabled\",\"subscriptionPolicies\":{\"locationPlacementId\":\"Public_2014-09-01\",\"quotaId\":\"PayAsYouGo_2014-09-01\",\"spendingLimit\":\"Off\"}}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "cache-control",
              "value": "no-cache"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "expires",
              "value": "-1"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "x-ms-ratelimit-remaining-tenant-reads",
              "value": "11999"
            },
            {
              "name": "x-ms-request-id",
              "value": "12e00ef9-a78a-4d11-8fca-976723c0487a"
            },
            {
              "name": "x-ms-correlation-request-id",
              "value": "12e00ef9-a78a-4d11-8fca-976723c0487a"
            },
            {
              "name": "x-ms-routing-request-id",
              "value": "CENTRALUS:20201019T171353Z:12e00ef9-a78a-4d11-8fca-976723c0487a"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "date",
              "value": "Mon, 19 Oct 2020 17:13:53 GMT"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "content-length",
              "value": "358"
            }
          ],
          "headersSize": 584,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2020-10-19T17:13:53.470Z",
        "time": 336,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 336
        }
      },
      {
        "_id": "94f3ecd46659c237327cab7c9352dce9",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "_fromType": "array",
              "name": "accept-language",
              "value": "en-US"
            },
            {
              "_fromType": "array",
              "name": "x-ms-client-request-id",
              "value": "9d48a56a-29c6-4c31-8d99-97ba6d5681b8"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "@azure/arm-storage/15.1.0 ms-rest-azure-js/2.0.1 ms-rest-js/2.0.7 Node/v14.9.0 OS/(x64-Darwin-19.6.0)"
            },
            {
              "_fromType": "array",
              "name": "cookie",
              "value": ""
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "management.azure.com"
            }
          ],
          "headersSize": 1881,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "api-version",
              "value": "2019-06-01"
            }
          ],
          "url": "https://management.azure.com/subscriptions/d3803fd6-2ba4-4286-80aa-f3d613ad59a7/resourceGroups/j1dev/providers/Microsoft.Storage/storageAccounts/ndowmon1j1devblobstorage/queueServices/default/queues?api-version=2019-06-01"
        },
        "response": {
          "bodySize": 102,
          "content": {
            "mimeType": "application/json",
            "size": 102,
            "text": "{\"error\":{\"code\":\"FeatureNotSupportedForAccount\",\"message\":\"Queue is not supported for the account.\"}}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "cache-control",
              "value": "no-cache"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "content-length",
              "value": "102"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "expires",
              "value": "-1"
            },
            {
              "name": "x-ms-request-id",
              "value": "20651871-8e22-49f5-949c-a6d53a7a5bbf"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-ms-ratelimit-remaining-subscription-reads",
              "value": "11999"
            },
            {
              "name": "server",
              "value": "Microsoft-Azure-Storage-Resource-Provider/1.0,Microsoft-HTTPAPI/2.0 Microsoft-HTTPAPI/2.0"
            },
            {
              "name": "x-ms-correlation-request-id",
              "value": "2a09c4a8-2fe2-4a7c-9ec6-dc725bd027d3"
            },
            {
              "name": "x-ms-routing-request-id",
              "value": "CENTRALUS:20201019T171354Z:2a09c4a8-2fe2-4a7c-9ec6-dc725bd027d3"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "date",
              "value": "Mon, 19 Oct 2020 17:13:53 GMT"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 627,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 400,
          "statusText": "Bad Request"
        },
        "startedDateTime": "2020-10-19T17:13:53.815Z",
        "time": 286,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 286
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
