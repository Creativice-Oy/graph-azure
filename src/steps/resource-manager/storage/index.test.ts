import { fetchStorageQueues } from '.';
import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { IntegrationConfig } from '../../../types';
import { setupAzureRecording } from '../../../../test/helpers/recording';
let recording: Recording;

afterEach(async () => {
  if (recording) {
    await recording.stop();
  }
});

test('step - storage queues', async () => {
  const instanceConfig: IntegrationConfig = {
    clientId: process.env.CLIENT_ID || 'clientId',
    clientSecret: process.env.CLIENT_SECRET || 'clientSecret',
    directoryId: '992d7bbe-b367-459c-a10f-cf3fd16103ab',
    subscriptionId: 'd3803fd6-2ba4-4286-80aa-f3d613ad59a7',
  };

  recording = setupAzureRecording({
    directory: __dirname,
    name: 'resource-manager-step-storage-queues',
    options: {
      recordFailedRequests: true,
    },
  });

  const context = createMockStepExecutionContext<IntegrationConfig>({
    instanceConfig,
    entities: [
      {
        _class: ['Service'],
        _type: 'azure_storage_account',
        _key: `/subscriptions/dccea45f-7035-4a17-8731-1fd46aaa74a0/resourceGroups/j1dev/providers/Microsoft.Storage/storageAccounts/ndowmon1j1dev`,
        id: `/subscriptions/dccea45f-7035-4a17-8731-1fd46aaa74a0/resourceGroups/j1dev/providers/Microsoft.Storage/storageAccounts/ndowmon1j1dev`,
        name: `ndowmon1j1dev`,
      },
      {
        _class: ['Service'],
        _type: 'azure_storage_account',
        _key: `/subscriptions/dccea45f-7035-4a17-8731-1fd46aaa74a0/resourceGroups/j1dev/providers/Microsoft.Storage/storageAccounts/ndowmon1j1devblobstorage`,
        id: `/subscriptions/dccea45f-7035-4a17-8731-1fd46aaa74a0/resourceGroups/j1dev/providers/Microsoft.Storage/storageAccounts/ndowmon1j1devblobstorage`,
        name: `ndowmon1j1devblobstorage`,
      },
      {
        _class: ['Group'],
        _type: 'azure_resource_group',
        _key: `/subscriptions/dccea45f-7035-4a17-8731-1fd46aaa74a0/resourceGroups/j1dev`,
      },
    ],
  });

  context.jobState.getData = jest.fn().mockResolvedValue({
    defaultDomain: 'www.fake-domain.com',
  });

  await fetchStorageQueues(context);

  expect(context.jobState.collectedEntities.length).toBeGreaterThan(0);
  expect(context.jobState.collectedEntities).toMatchGraphObjectSchema({
    _class: 'Queue',
    schema: {
      additionalProperties: false,
      properties: {
        _type: { const: 'azure_storage_queue' },
        _key: { type: 'string' },
        _class: { type: 'array', items: { const: 'Queue' } },
        id: { type: 'string' },
        name: { type: 'string' },
        displayName: { type: 'string' },
        type: { type: 'string' },
        resourceGroup: { type: 'string' },
        encrypted: { type: 'boolean' },
        _rawData: { type: 'array', items: { type: 'object' } },
      },
    },
  });
  expect(context.jobState.collectedRelationships.length).toEqual(1);
  expect(context.jobState.collectedRelationships).toEqual([
    {
      _key:
        '/subscriptions/dccea45f-7035-4a17-8731-1fd46aaa74a0/resourceGroups/j1dev/providers/Microsoft.Storage/storageAccounts/ndowmon1j1dev|has|/subscriptions/d3803fd6-2ba4-4286-80aa-f3d613ad59a7/resourceGroups/j1dev/providers/Microsoft.Storage/storageAccounts/ndowmon1j1dev/queueServices/default/queues/j1dev',
      _type: 'azure_storage_account_has_queue',
      _class: 'HAS',
      _fromEntityKey:
        '/subscriptions/dccea45f-7035-4a17-8731-1fd46aaa74a0/resourceGroups/j1dev/providers/Microsoft.Storage/storageAccounts/ndowmon1j1dev',
      _toEntityKey:
        '/subscriptions/d3803fd6-2ba4-4286-80aa-f3d613ad59a7/resourceGroups/j1dev/providers/Microsoft.Storage/storageAccounts/ndowmon1j1dev/queueServices/default/queues/j1dev',
      displayName: 'HAS',
    },
  ]);
});
