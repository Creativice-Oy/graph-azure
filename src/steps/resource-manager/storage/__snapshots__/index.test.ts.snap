// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rm-storage-accounts step 1`] = `
Array [
  Object {
    "_class": Array [
      "Service",
    ],
    "_key": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/DefaultResourceGroup-CUS/providers/Microsoft.Storage/storageAccounts/regenerateaccesskeytest",
    "_rawData": Array [
      Object {
        "name": "default",
        "rawData": Object {
          "accessTier": "Hot",
          "allowBlobPublicAccess": true,
          "creationTime": 2021-09-20T15:02:41.564Z,
          "enableHttpsTrafficOnly": true,
          "encryption": Object {
            "keySource": "Microsoft.Storage",
            "services": Object {
              "blob": Object {
                "enabled": true,
                "keyType": "Account",
                "lastEnabledTime": 2021-09-20T15:02:41.704Z,
              },
              "file": Object {
                "enabled": true,
                "keyType": "Account",
                "lastEnabledTime": 2021-09-20T15:02:41.704Z,
              },
            },
          },
          "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/DefaultResourceGroup-CUS/providers/Microsoft.Storage/storageAccounts/regenerateaccesskeytest",
          "kind": "StorageV2",
          "location": "eastus",
          "minimumTlsVersion": "TLS1_2",
          "name": "regenerateaccesskeytest",
          "networkRuleSet": Object {
            "bypass": "AzureServices",
            "defaultAction": "Allow",
            "ipRules": Array [],
            "virtualNetworkRules": Array [],
          },
          "primaryEndpoints": Object {
            "blob": "https://regenerateaccesskeytest.blob.core.windows.net/",
            "web": "https://regenerateaccesskeytest.z13.web.core.windows.net/",
          },
          "primaryLocation": "eastus",
          "privateEndpointConnections": Array [],
          "provisioningState": "Succeeded",
          "sku": Object {
            "name": "Premium_LRS",
            "tier": "Premium",
          },
          "statusOfPrimary": "available",
          "tags": Object {},
          "type": "Microsoft.Storage/storageAccounts",
        },
      },
    ],
    "_type": "azure_storage_account",
    "allowBlobPublicAccess": true,
    "blobAnalyticsLoggingDeleteEnabled": undefined,
    "blobAnalyticsLoggingReadEnabled": undefined,
    "blobAnalyticsLoggingWriteEnabled": undefined,
    "blobSoftDeleteEnabled": undefined,
    "blobSoftDeleteRetentionDays": undefined,
    "category": Array [
      "infrastructure",
    ],
    "createdOn": 1632150161564,
    "displayName": "regenerateaccesskeytest",
    "enableHttpsTrafficOnly": true,
    "encryptedBlob": true,
    "encryptedFileShare": true,
    "encryptedQueue": undefined,
    "encryptedTable": undefined,
    "encryption.keySource": "Microsoft.Storage",
    "endpoints": Array [
      "https://regenerateaccesskeytest.blob.core.windows.net/",
      "https://regenerateaccesskeytest.z13.web.core.windows.net/",
    ],
    "function": Array [
      "storage",
    ],
    "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/DefaultResourceGroup-CUS/providers/Microsoft.Storage/storageAccounts/regenerateaccesskeytest",
    "isAccessKeyRegenerated": true,
    "kind": "StorageV2",
    "name": "regenerateaccesskeytest",
    "networkRuleSet.allowedIpAddresses": "",
    "networkRuleSet.bypass": "AzureServices",
    "networkRuleSet.defaultAction": "Allow",
    "queueAnalyticsLoggingDeleteEnabled": undefined,
    "queueAnalyticsLoggingReadEnabled": undefined,
    "queueAnalyticsLoggingWriteEnabled": undefined,
    "region": "eastus",
    "resourceGroup": "defaultresourcegroup-cus",
    "sku": "Premium_LRS",
    "webLink": "https://portal.azure.com/#@j1AzureIntegrationDev.onmicrosoft.com/resource/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/DefaultResourceGroup-CUS/providers/Microsoft.Storage/storageAccounts/regenerateaccesskeytest",
  },
  Object {
    "_class": Array [
      "Service",
    ],
    "_key": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_Storage_Resources/providers/Microsoft.Storage/storageAccounts/stefanexamplestorage",
    "_rawData": Array [
      Object {
        "name": "default",
        "rawData": Object {
          "accessTier": "Hot",
          "allowBlobPublicAccess": true,
          "creationTime": 2021-09-07T13:14:18.446Z,
          "enableHttpsTrafficOnly": true,
          "encryption": Object {
            "keySource": "Microsoft.Storage",
            "services": Object {
              "blob": Object {
                "enabled": true,
                "keyType": "Account",
                "lastEnabledTime": 2021-09-07T13:14:18.555Z,
              },
              "file": Object {
                "enabled": true,
                "keyType": "Account",
                "lastEnabledTime": 2021-09-07T13:14:18.555Z,
              },
            },
          },
          "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_Storage_Resources/providers/Microsoft.Storage/storageAccounts/stefanexamplestorage",
          "kind": "StorageV2",
          "location": "eastus",
          "minimumTlsVersion": "TLS1_2",
          "name": "stefanexamplestorage",
          "networkRuleSet": Object {
            "bypass": "AzureServices",
            "defaultAction": "Allow",
            "ipRules": Array [],
            "virtualNetworkRules": Array [],
          },
          "primaryEndpoints": Object {
            "blob": "https://stefanexamplestorage.blob.core.windows.net/",
            "dfs": "https://stefanexamplestorage.dfs.core.windows.net/",
            "file": "https://stefanexamplestorage.file.core.windows.net/",
            "queue": "https://stefanexamplestorage.queue.core.windows.net/",
            "table": "https://stefanexamplestorage.table.core.windows.net/",
            "web": "https://stefanexamplestorage.z13.web.core.windows.net/",
          },
          "primaryLocation": "eastus",
          "privateEndpointConnections": Array [],
          "provisioningState": "Succeeded",
          "secondaryEndpoints": Object {
            "blob": "https://stefanexamplestorage-secondary.blob.core.windows.net/",
            "dfs": "https://stefanexamplestorage-secondary.dfs.core.windows.net/",
            "queue": "https://stefanexamplestorage-secondary.queue.core.windows.net/",
            "table": "https://stefanexamplestorage-secondary.table.core.windows.net/",
            "web": "https://stefanexamplestorage-secondary.z13.web.core.windows.net/",
          },
          "secondaryLocation": "westus",
          "sku": Object {
            "name": "Standard_RAGRS",
            "tier": "Standard",
          },
          "statusOfPrimary": "available",
          "statusOfSecondary": "available",
          "tags": Object {},
          "type": "Microsoft.Storage/storageAccounts",
        },
      },
      Object {
        "name": "blobServiceProperties",
        "rawData": Object {
          "blobAnalyticsLogging": Object {
            "deleteProperty": false,
            "read": false,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
            "write": false,
          },
          "cors": Array [],
          "defaultServiceVersion": undefined,
          "deleteRetentionPolicy": Object {
            "AllowPermanentDelete": "false",
            "days": 7,
            "enabled": true,
          },
          "hourMetrics": Object {
            "enabled": true,
            "includeAPIs": true,
            "retentionPolicy": Object {
              "days": 7,
              "enabled": true,
            },
            "version": "1.0",
          },
          "minuteMetrics": Object {
            "enabled": false,
            "includeAPIs": undefined,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
          },
          "staticWebsite": Object {
            "defaultIndexDocumentPath": undefined,
            "enabled": false,
            "errorDocument404Path": undefined,
            "indexDocument": undefined,
          },
        },
      },
      Object {
        "name": "queueServiceProperties",
        "rawData": Object {
          "cors": Array [],
          "hourMetrics": Object {
            "enabled": true,
            "includeAPIs": true,
            "retentionPolicy": Object {
              "days": 7,
              "enabled": true,
            },
            "version": "1.0",
          },
          "minuteMetrics": Object {
            "enabled": false,
            "includeAPIs": undefined,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
          },
          "queueAnalyticsLogging": Object {
            "deleteProperty": false,
            "read": false,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
            "write": false,
          },
        },
      },
    ],
    "_type": "azure_storage_account",
    "allowBlobPublicAccess": true,
    "blobAnalyticsLoggingDeleteEnabled": false,
    "blobAnalyticsLoggingReadEnabled": false,
    "blobAnalyticsLoggingWriteEnabled": false,
    "blobSoftDeleteEnabled": true,
    "blobSoftDeleteRetentionDays": 7,
    "category": Array [
      "infrastructure",
    ],
    "createdOn": 1631020458446,
    "displayName": "stefanexamplestorage",
    "enableHttpsTrafficOnly": true,
    "encryptedBlob": true,
    "encryptedFileShare": true,
    "encryptedQueue": undefined,
    "encryptedTable": undefined,
    "encryption.keySource": "Microsoft.Storage",
    "endpoints": Array [
      "https://stefanexamplestorage.blob.core.windows.net/",
      "https://stefanexamplestorage.queue.core.windows.net/",
      "https://stefanexamplestorage.table.core.windows.net/",
      "https://stefanexamplestorage.file.core.windows.net/",
      "https://stefanexamplestorage.z13.web.core.windows.net/",
      "https://stefanexamplestorage.dfs.core.windows.net/",
    ],
    "function": Array [
      "storage",
    ],
    "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_Storage_Resources/providers/Microsoft.Storage/storageAccounts/stefanexamplestorage",
    "isAccessKeyRegenerated": false,
    "kind": "StorageV2",
    "name": "stefanexamplestorage",
    "networkRuleSet.allowedIpAddresses": "",
    "networkRuleSet.bypass": "AzureServices",
    "networkRuleSet.defaultAction": "Allow",
    "queueAnalyticsLoggingDeleteEnabled": false,
    "queueAnalyticsLoggingReadEnabled": false,
    "queueAnalyticsLoggingWriteEnabled": false,
    "region": "eastus",
    "resourceGroup": "stefan_storage_resources",
    "sku": "Standard_RAGRS",
    "webLink": "https://portal.azure.com/#@j1AzureIntegrationDev.onmicrosoft.com/resource/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_Storage_Resources/providers/Microsoft.Storage/storageAccounts/stefanexamplestorage",
  },
  Object {
    "_class": Array [
      "Service",
    ],
    "_key": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_AppServices_Group/providers/Microsoft.Storage/storageAccounts/storageaccountstefaafb8",
    "_rawData": Array [
      Object {
        "name": "default",
        "rawData": Object {
          "creationTime": 2021-09-16T14:50:27.037Z,
          "enableHttpsTrafficOnly": true,
          "encryption": Object {
            "keySource": "Microsoft.Storage",
            "services": Object {
              "blob": Object {
                "enabled": true,
                "keyType": "Account",
                "lastEnabledTime": 2021-09-16T14:50:27.130Z,
              },
              "file": Object {
                "enabled": true,
                "keyType": "Account",
                "lastEnabledTime": 2021-09-16T14:50:27.130Z,
              },
            },
          },
          "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_AppServices_Group/providers/Microsoft.Storage/storageAccounts/storageaccountstefaafb8",
          "kind": "Storage",
          "location": "centralus",
          "minimumTlsVersion": "TLS1_2",
          "name": "storageaccountstefaafb8",
          "networkRuleSet": Object {
            "bypass": "AzureServices",
            "defaultAction": "Allow",
            "ipRules": Array [],
            "virtualNetworkRules": Array [],
          },
          "primaryEndpoints": Object {
            "blob": "https://storageaccountstefaafb8.blob.core.windows.net/",
            "file": "https://storageaccountstefaafb8.file.core.windows.net/",
            "queue": "https://storageaccountstefaafb8.queue.core.windows.net/",
            "table": "https://storageaccountstefaafb8.table.core.windows.net/",
          },
          "primaryLocation": "centralus",
          "privateEndpointConnections": Array [],
          "provisioningState": "Succeeded",
          "sku": Object {
            "name": "Standard_LRS",
            "tier": "Standard",
          },
          "statusOfPrimary": "available",
          "tags": Object {},
          "type": "Microsoft.Storage/storageAccounts",
        },
      },
      Object {
        "name": "blobServiceProperties",
        "rawData": Object {
          "blobAnalyticsLogging": Object {
            "deleteProperty": false,
            "read": false,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
            "write": false,
          },
          "cors": Array [],
          "defaultServiceVersion": undefined,
          "deleteRetentionPolicy": Object {
            "AllowPermanentDelete": "false",
            "days": undefined,
            "enabled": false,
          },
          "hourMetrics": Object {
            "enabled": true,
            "includeAPIs": true,
            "retentionPolicy": Object {
              "days": 7,
              "enabled": true,
            },
            "version": "1.0",
          },
          "minuteMetrics": Object {
            "enabled": false,
            "includeAPIs": undefined,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
          },
          "staticWebsite": undefined,
        },
      },
      Object {
        "name": "queueServiceProperties",
        "rawData": Object {
          "cors": Array [],
          "hourMetrics": Object {
            "enabled": true,
            "includeAPIs": true,
            "retentionPolicy": Object {
              "days": 7,
              "enabled": true,
            },
            "version": "1.0",
          },
          "minuteMetrics": Object {
            "enabled": false,
            "includeAPIs": undefined,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
          },
          "queueAnalyticsLogging": Object {
            "deleteProperty": false,
            "read": false,
            "retentionPolicy": Object {
              "days": undefined,
              "enabled": false,
            },
            "version": "1.0",
            "write": false,
          },
        },
      },
    ],
    "_type": "azure_storage_account",
    "allowBlobPublicAccess": undefined,
    "blobAnalyticsLoggingDeleteEnabled": false,
    "blobAnalyticsLoggingReadEnabled": false,
    "blobAnalyticsLoggingWriteEnabled": false,
    "blobSoftDeleteEnabled": false,
    "blobSoftDeleteRetentionDays": undefined,
    "category": Array [
      "infrastructure",
    ],
    "createdOn": 1631803827037,
    "displayName": "storageaccountstefaafb8",
    "enableHttpsTrafficOnly": true,
    "encryptedBlob": true,
    "encryptedFileShare": true,
    "encryptedQueue": undefined,
    "encryptedTable": undefined,
    "encryption.keySource": "Microsoft.Storage",
    "endpoints": Array [
      "https://storageaccountstefaafb8.blob.core.windows.net/",
      "https://storageaccountstefaafb8.queue.core.windows.net/",
      "https://storageaccountstefaafb8.table.core.windows.net/",
      "https://storageaccountstefaafb8.file.core.windows.net/",
    ],
    "function": Array [
      "storage",
    ],
    "id": "/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_AppServices_Group/providers/Microsoft.Storage/storageAccounts/storageaccountstefaafb8",
    "isAccessKeyRegenerated": false,
    "kind": "Storage",
    "name": "storageaccountstefaafb8",
    "networkRuleSet.allowedIpAddresses": "",
    "networkRuleSet.bypass": "AzureServices",
    "networkRuleSet.defaultAction": "Allow",
    "queueAnalyticsLoggingDeleteEnabled": false,
    "queueAnalyticsLoggingReadEnabled": false,
    "queueAnalyticsLoggingWriteEnabled": false,
    "region": "centralus",
    "resourceGroup": "stefan_appservices_group",
    "sku": "Standard_LRS",
    "webLink": "https://portal.azure.com/#@j1AzureIntegrationDev.onmicrosoft.com/resource/subscriptions/193f89dc-6225-4a80-bacb-96b32fbf6dd0/resourceGroups/Stefan_AppServices_Group/providers/Microsoft.Storage/storageAccounts/storageaccountstefaafb8",
  },
]
`;
